### PR TITLE
#162681773 Add create account endpoint 

### PR DESCRIPTION
**Pull request by uthdev**
#### What does this PR do?
This PR Adds the signup endpoint
#### Description of Task to be completed?
Have the POST /api/v1/auth/signup working
Create route and controller for signup
validate user input
Hash user password 
check if user already exist
persist user data if user doesn't exist
#### How should this be manually tested?
After cloning the repo and cd into it, run npm install, npm start
open postman and navigate to /users/signup
Create an account then navigate to /user/login
input email and password used to create account
#### Questions:

### CODE REVIEW

File: server/src/config/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const { Pool } = require('pg')
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+console.log(process.env.DATABASE_URL);
```
Comment (sannytee) [inline server/src/config/index.js]: Remove the console.log

File: server/src/config/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const pool = new Pool({ 
+  connectionString : process.env.DATABASE_URL,
+})
+
+module.exports = {
```
Comment (sannytee) [inline server/src/config/index.js]: why are you making use of es5 export instead of es6?

File: server/src/helpers/responses.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const responses = {};
+
+responses.errorProcessing = (req, res) => {
+  return res.status(500).json({
```
Comment (sannytee) [inline server/src/helpers/responses.js:4]: The status code doesn't really match the error message. 5xx are basically for server error.

**Review by sannytee — CHANGES_REQUESTED** [review]
Remove all console.log statement

File: server/src/config/index.js (hunk 1 above)
Comment (uthdev) [reply server/src/config/index.js]: Done. Thank you

File: server/src/helpers/responses.js (hunk 3 above)
Comment (uthdev) [reply server/src/helpers/responses.js:4]: Resolved

**Review by uthdev — COMMENTED** [review]
I've implemented the changes. Kindly check them out

File: server/src/config/index.js (hunk 2 above)
Comment (uthdev) [reply server/src/config/index.js]: Changes made

**Pull request merged**